### PR TITLE
Remove outdated copy of native binaries

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

https://github.com/microsoft/appcenter-sdk-react-native/tree/develop/AppCenterReactNativeShared/ios

*.frameworks are used here only for core headers. Also, it looks very outdated.
We need to drop the copy of it and use the latest version from Pods like in other places

## Related PRs or issues

[AB#66766](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/66766)

### CODE REVIEW

**Comment by russelarms** [comment]
I have added an additional pipeline step to install Pods (it's my testing pipeline):
https://dev.azure.com/msmobilecenter/SDK/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=3531
It should be added to the main pipeline after the approval.

File: AppCenterReactNativeShared/ios/post-clone.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+echo "Installing pods for AppCenterReactNativeShared in `pwd`"
+
```
Comment (dhei) [inline AppCenterReactNativeShared/ios/post-clone.sh]: Please add `pod repo update` before `pod install`.

Azure Pipeline mac VM doesn't always come with the latest pod repos.

File: AppCenterReactNativeShared/ios/Podfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Uncomment the next line to define a global platform for your project
+# platform :ios, '9.0'
+
+target 'AppCenterReactNativeShared' do
+  # Comment the next line if you don't want to use dynamic frameworks
+  use_frameworks!
```
Comment (dhei) [inline AppCenterReactNativeShared/ios/Podfile]: Are we sure we want to use dynamic frameworks here? 

Does `use_frameworks!` matter here? If you comment out `use_frameworks!`, would it still work?

I don't see `use_framework!` in TestApp and DemoApp, so I am not sure if we should or shouldn't use it.

File: AppCenterReactNativeShared/ios/Podfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+  # Pods for AppCenterReactNativeShared
+  pod 'AppCenter'
```
Comment (dhei) [inline AppCenterReactNativeShared/ios/Podfile]: We use `pod 'AppCenter', '~> 2.3.0'` with a version number in DemoApp to allow testing pre-release private native ios sdk. It's more flexible that way. Maybe we should do the same here.

File: AppCenterReactNativeShared/ios/Podfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Uncomment the next line to define a global platform for your project
```
Comment (dhei) [inline AppCenterReactNativeShared/ios/Podfile]: Just nit-picking, I think we can get rid of these two lines of comments as we won't need it for global scope anyway.

File: AppCenterReactNativeShared/ios/Podfile (hunk 2 above)
Comment (russelarms) [reply AppCenterReactNativeShared/ios/Podfile]: It doesn't build without this line, here are the errors I got:

```
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: can't locate file for: -lPods-AppCenterReactNativeShared

error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: -lPods-AppCenterReactNativeShared is not an object file (not allowed in a library)
```

**Pull request merged**